relay-server/src/utils/envelope_context.rs

```diff
@@ -72,15 +73,21 @@ pub struct EnvelopeContext {
 
 impl EnvelopeContext {
     /// Computes an envelope context from the given envelope.
-    fn new_internal(envelope: &Envelope, slot: Option<SemaphorePermit>) -> Self {
+    fn new_internal(envelope: Box<Envelope>, slot: Option<SemaphorePermit>) -> Self {
         let meta = &envelope.meta();
+        let summary = EnvelopeSummary::compute(envelope.as_ref());
+        let event_id = envelope.event_id();
+        let start_time = meta.start_time();
+        let remote_addr = meta.client_addr();
+        let scoping = meta.get_partial_scoping();
         Self {
-            summary: EnvelopeSummary::compute(envelope),
-            start_time: meta.start_time(),
-            received_at: relay_common::instant_to_date_time(meta.start_time()),
-            event_id: envelope.event_id(),
-            remote_addr: meta.client_addr(),
-            scoping: meta.get_partial_scoping(),
+            envelope,
+            summary,
+            start_time,
+            received_at: relay_common::instant_to_date_time(start_time),
+            event_id,
+            remote_addr,
+            scoping,
```

Comment:
This is personal preference, but in this case at least, I prefer the previous approach instead of having variables. However, if you decide to leave the variables, I'd also extract `received_at`.

Reply:
I need the variables because the envelope is being moved now, so I cannot reference `meta` anymore. But good catch, will also make a variable for `received_at`.